.github/workflows/backport-queue.yml

```diff
@@ -0,0 +1,169 @@
+            --filter-release --format=sha --reverse | while read -r COMMIT ; do
+            if backport "$COMMIT" 2>&1 >output; then
+              MAX_COMMITS="$((MAX_COMMITS-1))"
+            else
+              {
+                EOF="$(node -p 'crypto.randomUUID()')"
+                echo "$COMMIT<<$EOF"
+                echo "…"
+                tail output # Cutting the output to avoid exceeding memory.
+                echo "$EOF"
+              } >> "$GITHUB_OUTPUT"
+              git cherry-pick --skip || git reset HEAD^ --hard
+            fi
+            cat output
+            [ "$MAX_COMMITS" = "0" ] && break || true
+          done
+          rm -f output
+        env:
+          MAX_COMMITS: 99 # backport commits 99 at a time to limit the risk of timeout
+          CURRENT_RELEASE_LINE: ${{ matrix.branch.release-line }}
+          BASE_REF: origin/${{ matrix.branch.base-branch }}
+          BACKPORT_BRANCH: ${{ matrix.branch.release-line }}-staging-auto-backport
+
+      - name: Get local HEAD
+        id: head
+        run: echo "HEAD=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
+
+      - name: Push to working branch
+        if: ${{ steps.fetch.outputs.WORKING_BRANCH_TIP != steps.head.outputs.HEAD }}
+        run: git push origin "HEAD:refs/heads/$BACKPORT_BRANCH" --force
+        env:
+          BACKPORT_BRANCH: ${{ matrix.branch.release-line }}-staging-auto-backport
+
+      - name: Report errors
+        if: ${{
+            steps.fetch.outputs.WORKING_BRANCH_TIP != steps.head.outputs.HEAD &&
+            toJson(steps.queue.outputs) != '{}'
+            }}
+        run: |
+          node -<<'EOF' | gh issue create --repo "$GITHUB_REPOSITORY_OWNER/Release"\
+            -t "Autobackport is failing on $CURRENT_RELEASE_LINE" \
+            --label 'Release-agenda' --body-file -
+          console.log(`Workflow URL: <${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}>`);
+          console.log('\n<ul>');
+          for (const [sha, output] of Object.entries(JSON.parse(process.env.FAILED_BACKPORTS))) {
+            console.log('<li><details>');
+            console.log(`<summary><code>${sha}</code></summary>`);
+            console.log(`\n${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/commit/${sha}\n\n${'```'}\n${output}\n${'```'}\n`);
+            console.log('</details></li>');
+          }
+          console.log('</ul>');
+          console.log('Try to cherry-pick the commit manually or add `backport-requested` label and leave a comment on the associated PR.')
+          EOF
+        env:
+          GH_TOKEN: ${{ secrets.GH_USER_TOKEN }}
+          FAILED_BACKPORTS: ${{ toJson(steps.queue.outputs) }}
+          CURRENT_RELEASE_LINE: ${{ matrix.branch.release-line }}
```

Comment:
This should be replaced with https://github.com/nodejs/node-core-utils/pull/875 once that lands, it has many benefits:
- lives in `ncu` which is a more friendly developer tooling space for the majority of collaborators and might be simpler to be unit tested. Note that [this has been requested when I took a stab at it earlier this year](https://github.com/nodejs/node/pull/51916#issuecomment-1971084782)
- automatically labels (and can optionally comment) the original PR that needs manual backporting
- generates a report (in multiple formats) to the releaser of successful and failed cherry-pick commits (wip: open a new issue with the report so that releasers can follow up with manually cherry-picking simpler conflicts or automate ask the original PR author for backport)
- support to pagination, so that the releaser can manually limit the amount of commits to be cherry-picked
- validates many of the steps along the way so that the script is more resilient to unexpected API failures
- (when run locally) stores the state of the current cherry-pick progress and its result so that a releaser can run the command again in order to resume in the case of an unexpected failure (not sure if that state persistency could be applied to GitHub Actions)
- automatically figures out the `base-branch` based on the current working staging branch
- better fine control, so it's possible for releasers to have a dry-run by using options such as `--skipGH` to avoid any GitHub API interaction, this way it's possible to see the results in your machine, run tests, etc before proceeding
- more complex logic, such as skipping commits that belong to a PR that was already marked as backport-requested

Reply:
We don't want to label not comment automatically the PRs before first trying to manually resolve the conflicts ourselves, otherwise it's going to be much too noisy for it to be worth it

Reply:
yes, we don't want to comment automatically as it's been discussed in https://github.com/nodejs/node-core-utils/pull/875 already.

That said, I'm not aware of any consensus around avoiding automatic labelling, labels are a great way to communicate and create process around managing these commits that will not create any notification spamming.

Reply:
This makes much more sense after our discussion on the Release WG meeting since now I see that the goal here is not to replace / suplement the idea of landing commits on staging (similar to `git node staging`) but rather have an early validation on what commits are landing on the staging branch vs conflicting.